packages/drawer/src/views/modern/Drawer.tsx

```diff
@@ -378,11 +379,12 @@ export default function Drawer({
                   toggleDrawer({ open: false, isUserInitiated: true })
                 }
                 style={overlayStyle}
+                accessibilityLabel={closeAccessibilityLabel}
               />
             ) : null}
           </Animated.View>
           <Animated.View
-            accessibilityViewIsModal={isOpen && drawerType !== 'permanent'}
+            accessibilityViewIsModal={!isOpen}
```

Comment:
now, for iOS (accessibilityViewIsModal is iOS only prop), no matter the type of drawer, it will always allow the overlay to be accessible tappable when the drawer is open.

before, it would only allow the overlay to be tappable if it wasn't open for the permanent type (which doesn't make much sense)

Reply:
I'm not sure what do you mean since the overlay isn't (or shouldn't be) rendered for permanent drawer.

Reply:
This logic does two things: 

1) Eliminates the dependancy on permanent drawerType. As overlay is not present when type == permanent, I don't see a reason to have a second check here. In other words, overlay specific logic can ignore the permanent drawerType altogether because permanent type does not even implement overlay.

2) Allows the overlay to be accessibility selectable (ie readable by VoiceOver) when the drawer is open in iOS.

Reply:
The problem with this logic is that with permanent drawer, the open and closed state of the drawer doesn't matter. It's always present and is interactable. So it needs to be treated differently. The modal part of the drawer only matters when it's not permanent.

Though reading the description of this prop, I think it shouldn't be used here at all. The a11y of the overlay should already be taken care of the props on it so there are no siblings to hide. The closed drawer also doesn't appear like a modal so the prop feels like a misuse.

Reply:
Agree! Getting rid of this prop altogether would work too!